dcos_launch/platforms/onprem.py

```diff
@@ -238,30 +238,51 @@ def do_postflight(client: ssh_client.AsyncSshClient):
     """ Runs a script that will check if DC/OS is operational without needing to authenticate
     """
     postflight_script = """
-T=900
-if [ -f /opt/mesosphere/etc/dcos-diagnostics-runner-config.json ]; then
+function run_command_until_success() {
+    # Run $@ until it exits 0 or until it has been tried 900 times. Prints shell output and returns the status of the
+    # last attempted run.
+    cmd=$@
+    max_runs=900
+
+    runs=$max_runs
+    until out=$($cmd) || [[ runs -eq 0 ]]; do
+        sleep 1
+        let runs=runs-1
+    done
+    retcode=$?
+
+    echo "$out"
```

Comment:
Would it be helpful to also print how many times it had to try before it worked?

Reply:
I think it's more useful to know how much time elapsed before checks started passing. That can be inferred from timestamps in logs, so I wasn't too worried about adding that info here.